app/models/user.rb

```diff
@@ -12,8 +14,39 @@ class User < ApplicationRecord
 
   has_secure_password
 
-  private
+  # Returns the hash digest of the given string.
+  def self.digest string
+    cost = if ActiveModel::SecurePassword.min_cost
+             BCrypt::Engine::MIN_COST
+           else
+             BCrypt::Engine.cost
+           end
+    BCrypt::Password.create(string, cost: cost)
+  end
+
+  # Returns a random token.
+  def self.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  # Remembers a user in the database for use in persistent sessions.
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
```

Comment:
```suggestion
    update_column :remember_digest, User.digest(remember_token)
```
ri sẽ hợp lý hơn nhé, vì chỗ này ko cần chạy callback